tcms/testruns/views.py

```diff
@@ -963,8 +972,8 @@ class AddCasesToRunView(View):
     template_name = 'run/assign_case.html'
 
     @method_decorator(permission_required(permission))
-    def dispatch(self, *args, **kwargs):
-        return super(AddCasesToRunView, self).dispatch(*args, **kwargs)
+    def dispatch(self, request, *args, **kwargs):
+        return super(AddCasesToRunView, self).dispatch(request, *args, **kwargs)
```

Comment:
This is a lef-over from older Django versions I think. Now you can decorate a class based view as follows:

```
@method_decorator(permission_required('testruns.change_testcaserun'), name='dispatch')
class UpdateCaseRunStatusView(View):
```

You don't need to override the dispatch() method only to decorate it.